spring-cloud-dataflow-aggregate-task/src/main/java/org/springframework/cloud/dataflow/aggregate/task/impl/AggregateDataFlowTaskExecutionQueryDao.java

```diff
@@ -86,75 +88,74 @@ public class AggregateDataFlowTaskExecutionQueryDao implements DataflowTaskExecu
 			" where PARENT_EXECUTION_ID IN (:taskExecutionIds)" +
 			" and (SELECT COUNT(*) FROM AGGREGATE_TASK_EXECUTION_PARAMS P " +
 			"           WHERE P.TASK_EXECUTION_ID=TASK_EXECUTION_ID " +
 			"             AND P.SCHEMA_TARGET=SCHEMA_TARGET" +
 			"             AND P.TASK_PARAM = :schemaTarget) > 0";
 
-	private static final String GET_EXECUTION_BY_EXTERNAL_EXECUTION_ID = "SELECT TASK_EXECUTION_ID,"
-			+ "START_TIME, END_TIME, TASK_NAME, EXIT_CODE,"
-			+ "EXIT_MESSAGE, ERROR_MESSAGE, LAST_UPDATED, EXTERNAL_EXECUTION_ID,"
-			+ "PARENT_EXECUTION_ID, SCHEMA_TARGET"
-			+ " from AGGREGATE_TASK_EXECUTION where EXTERNAL_EXECUTION_ID = :externalExecutionId and TASK_NAME = :taskName";
-
-	private static final String GET_EXECUTIONS_BY_NAME_COMPLETED = "SELECT TASK_EXECUTION_ID,"
-			+ "START_TIME, END_TIME, TASK_NAME, EXIT_CODE,"
-			+ "EXIT_MESSAGE, ERROR_MESSAGE, LAST_UPDATED, EXTERNAL_EXECUTION_ID,"
-			+ "PARENT_EXECUTION_ID, SCHEMA_TARGET"
-			+ " from AGGREGATE_TASK_EXECUTION where TASK_NAME = :taskName AND END_TIME IS NOT NULL";
-
-	private static final String GET_EXECUTIONS_BY_NAME = "SELECT TASK_EXECUTION_ID,"
-			+ "START_TIME, END_TIME, TASK_NAME, EXIT_CODE,"
-			+ "EXIT_MESSAGE, ERROR_MESSAGE, LAST_UPDATED, EXTERNAL_EXECUTION_ID,"
-			+ "PARENT_EXECUTION_ID, SCHEMA_TARGET"
-			+ " from AGGREGATE_TASK_EXECUTION where TASK_NAME = :taskName";
-	private static final String GET_EXECUTIONS_COMPLETED = "SELECT TASK_EXECUTION_ID,"
-			+ "START_TIME, END_TIME, TASK_NAME, EXIT_CODE,"
-			+ "EXIT_MESSAGE, ERROR_MESSAGE, LAST_UPDATED, EXTERNAL_EXECUTION_ID,"
-			+ "PARENT_EXECUTION_ID, SCHEMA_TARGET"
-			+ " from AGGREGATE_TASK_EXECUTION where END_TIME IS NOT NULL";
-
-	private static final String GET_EXECUTIONS = "SELECT TASK_EXECUTION_ID,"
-			+ "START_TIME, END_TIME, TASK_NAME, EXIT_CODE,"
-			+ "EXIT_MESSAGE, ERROR_MESSAGE, LAST_UPDATED, EXTERNAL_EXECUTION_ID,"
-			+ "PARENT_EXECUTION_ID, SCHEMA_TARGET"
-			+ " from AGGREGATE_TASK_EXECUTION";
+	private static final String GET_EXECUTION_BY_EXTERNAL_EXECUTION_ID = GET_EXECUTIONS +
+			" where EXTERNAL_EXECUTION_ID = :externalExecutionId and TASK_NAME = :taskName";
+
+	private static final String GET_EXECUTIONS_BY_NAME_COMPLETED = GET_EXECUTIONS +
+			" where TASK_NAME = :taskName AND END_TIME IS NOT NULL";
+
+	private static final String GET_EXECUTIONS_BY_NAME = GET_EXECUTIONS +
+			" where TASK_NAME = :taskName";
+
+	private static final String GET_EXECUTIONS_COMPLETED = GET_EXECUTIONS +
+			" where END_TIME IS NOT NULL";
+
+	private static final String GET_EXECUTION_BY_NAME_COMPLETED_BEFORE_END_TIME = GET_EXECUTIONS +
+			" where TASK_NAME = :taskName AND END_TIME IS NOT NULL AND END_TIME < :endTime";
+
+	private static final String GET_EXECUTIONS_COMPLETED_BEFORE_END_TIME = GET_EXECUTIONS +
+			" where END_TIME IS NOT NULL AND END_TIME < :endTime";
+
+	private static final String GET_EXECUTION_BY_NAME_BEFORE_END_TIME = GET_EXECUTIONS +
```

Comment:
@klopfdreh thanks for the updates. This looks much better! 

One point of contention before I merge though, these queries that include `END_TIME < :endTime` but do not make sure `END_TIME IS NOT NULL`... since we are using the end date as the criteria, what do we expect those w null END_TIME will do? Be included or not?

Reply:
Yes I see  - we can remove `GET_EXECUTIONS_BEFORE_END_TIME` and `GET_EXECUTION_BY_NAME_BEFORE_END_TIME` and remove the completed parameter from `findTaskExecutionsBeforeEndTime` and simplify the if statement a bit. I am going to do so later. 👍